### PR TITLE
🐍  Rename spike detector -> threshold detector

### LINKED ISSUE #1642: Rename spike_detector to something more descriptive
The term `spike_detector` is confusing. Proposed renamings:

* `membrane_threshold_detector`
* `mtd`

### PR DESCRIPTION
**Pull request by thorstenhater**
Rename `spike_detector` to `threshold_detector` and make `spike_detector` throw a deprecation error.
This brings the Python API more in line with C++ and is closer to the object's actual function.

Closes #1642

### CODE REVIEW

File: python/cells.cpp [hunk 1]
```diff
@@ -577,8 +577,13 @@ void register_cells(pybind11::module& m) {
...
     // arb::threshold_detector
+    struct spike_detector {};
+    pybind11::class_<spike_detector>(m, "spike_detector")
+        .def(pybind11::init<>([](pybind11::object) {
+            throw arb::arbor_exception{"spike_detector is deprecated, use threshold_detector instead."};
+        }));
 
-    pybind11::class_<arb::threshold_detector> detector(m, "spike_detector",
+    pybind11::class_<arb::threshold_detector> detector(m, "threshold_detector",
             "A spike detector, generates a spike when voltage crosses a threshold. Can be used as source endpoint for an arbor.connection.");
```
Comment (schmitts) [inline python/cells.cpp:587]: A **threshold** detector ...

File: python/example/plasticity.py [hunk 2]
```diff
@@ -88,11 +88,13 @@ def add_connection_to_spike_source(self, to):
 # Note that the connection is just _recorded_ in the recipe, the actual connectivity
 # is set up in the simulation construction.
 rec.add_connection_to_spike_source(1)
+
+# Now construct the simulation
 sim = A.simulation(rec)
```
Comment (schmitts) [inline python/example/plasticity.py:93]: remove drive by changes

**Review by schmitts — CHANGES_REQUESTED** [review]
We just break API? I'm fine with that.

File: python/cells.cpp (hunk 1 above)
Comment (thorstenhater) [reply python/cells.cpp:589]: It detects a threshold being crossed. A detector could be anything.

**Review by brenthuisman — APPROVED** [review]
Yes!!!!!!

**Comment by thorstenhater** [comment]
I appreciate the excitement for homogenising APIs, but any particular reason?

**Comment by brenthuisman** [comment]
This name has been confusing me for some time now.

**Pull request merged**